Source/JavaScriptCore/runtime/IteratorOperations.h

```
@@ -170,7 +184,14 @@ void forEachInIterable(JSGlobalObject& globalObject, JSObject* iterable, JSValue
     auto& vm = getVM(&globalObject);
     auto scope = DECLARE_THROW_SCOPE(vm);
 
-    if (getIterationMode(vm, &globalObject, iterable, iteratorMethod) == IterationMode::FastArray) {
+    auto validationResult = validateIterable(vm, iterable, iteratorMethod);
+    if (validationResult != IterableValidationResult::Valid) {
+        createIteratorError(&globalObject, vm, validationResult, iterable);
+        return;
```

Returning an error object is not aligned to JSC's convention.
If `scope` is defined and if it is throwing an error, you need to throw an error.

---

Fixed.